lnbits/extensions/boltz/crud.py

```diff
@@ -0,0 +1,227 @@
+from http import HTTPStatus
+from typing import List, Optional, Union
+
+from loguru import logger
+from starlette.exceptions import HTTPException
+
+from . import db
+from .models import (
+    CreateReverseSubmarineSwap,
+    CreateSubmarineSwap,
+    ReverseSubmarineSwap,
+    SubmarineSwap,
+)
+
+"""
+Submarine Swaps
+"""
+
+
+async def get_submarine_swaps(wallet_ids: Union[str, List[str]]) -> List[SubmarineSwap]:
+    if isinstance(wallet_ids, str):
+        wallet_ids = [wallet_ids]
+
+    q = ",".join(["?"] * len(wallet_ids))
+    rows = await db.fetchall(
+        f"SELECT * FROM boltz.submarineswap WHERE wallet IN ({q}) order by time DESC",
+        (*wallet_ids,),
+    )
+
+    return [SubmarineSwap(**row) for row in rows]
+
+
+async def get_pending_submarine_swaps(
+    wallet_ids: Union[str, List[str]]
+) -> List[SubmarineSwap]:
+    if isinstance(wallet_ids, str):
+        wallet_ids = [wallet_ids]
+
+    q = ",".join(["?"] * len(wallet_ids))
+    rows = await db.fetchall(
+        f"SELECT * FROM boltz.submarineswap WHERE wallet IN ({q}) and status='pending' order by time DESC",
+        (*wallet_ids,),
+    )
+    return [SubmarineSwap(**row) for row in rows]
+
+
+async def get_all_pending_submarine_swaps() -> List[SubmarineSwap]:
+    rows = await db.fetchall(
+        f"SELECT * FROM boltz.submarineswap WHERE status='pending' order by time DESC",
+    )
+    return [SubmarineSwap(**row) for row in rows]
+
+
+async def get_submarine_swap(swap_id) -> SubmarineSwap:
+    row = await db.fetchone(
+        "SELECT * FROM boltz.submarineswap WHERE id = ?", (swap_id,)
+    )
+    return SubmarineSwap(**row) if row else None
+
+
+async def create_submarine_swap(swap: SubmarineSwap) -> Optional[SubmarineSwap]:
+
+    await db.execute(
+        """
+        INSERT INTO boltz.submarineswap (
+            id,
+            wallet,
+            payment_hash,
+            status,
+            boltz_id,
+            refund_privkey,
+            refund_address,
+            expected_amount,
+            timeout_block_height,
+            address,
+            bip21,
+            redeem_script,
+            amount
+        )
+        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+        """,
+        (
+            swap.id,
+            swap.wallet,
+            swap.payment_hash,
+            swap.status,
+            swap.boltz_id,
+            swap.refund_privkey,
+            swap.refund_address,
+            swap.expected_amount,
+            swap.timeout_block_height,
+            swap.address,
+            swap.bip21,
+            swap.redeem_script,
+            swap.amount,
+        ),
+    )
+    return await get_submarine_swap(swap.id)
+
+
+async def delete_submarine_swap(swap_id):
```

Comment:
Unused method. And I don't think you should ever delete any kind of swap information from the database